File: Outdated/do_get_orbit.c
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>
#include <getopt.h>

#include "headers.h"

void usage(char *arg0) {
  printf("usage: %s [year] [month] [day] [hour] [minute] <second>\n", arg0);
}

int main(int argc, char **argv) {
  int year, month, day, hour, min, sec;
  double carlong, sun_ob[3];
  
  int opt;
  char optstring[] = "h";

  /* process command line arguments */
  while ((opt = getopt(argc, argv, optstring)) != -1) {
    switch (opt) {
    case 'h':
      usage(argv[0]);
      return 0;
      break;
    default:
      usage(argv[0]);
      return 1;
    }
  }

  if (argc - optind <= 4) {
    usage(argv[0]);
    return 1;
  } else if ((argc - optind) <= 6) {
    sscanf(argv[optind], "%d", &year);
    sscanf(argv[optind+1], "%d", &month);
    sscanf(argv[optind+2], "%d", &day);
    sscanf(argv[optind+3], "%d", &hour);
    sscanf(argv[optind+4], "%d", &min);

    if ((argc - optind) == 6) {
      sscanf(argv[optind+5], "%d", &sec);
    }
    else {
      sec = 0;
    }
  }
  else {
    usage(argv[0]);
    return 1;
  }

  get_orbit_wrapper(year, month, day, hour, min, sun_ob, &carlong);
  printf("%+g\t%g\t%g\t%g\n",  carlong, sun_ob[0], sun_ob[1], sun_ob[2]);

  return 0;
}

File: matlab/do_get_orbit.m
function [carlong, sun_ob] = do_get_orbit(year, month, day, hour, ...
                                          minute, second)
%[carlong, sun_ob] = do_get_orbit(year, month, day, hour, minute, [second])


if (nargin == 5)
  second = 0;
end

pd = pwd;
cd('~/src/srt/');

command = ['./do_get_orbit ', ...
           num2str(year), ' ', ...
           num2str(month), ' ', ...
           num2str(day), ' ', ...
           num2str(hour), ' ', ...
           num2str(minute), ' ', ...
           num2str(second)];
    
[s, stdout] = my_system(command);

cd(pd);

orbit_info = sscanf(stdout, '%g\t%g\t%g\t%g');
    
if (length(orbit_info) ~= 4)
  error('Maformed data retuned by do_get_orbit');
end
    
carlong = orbit_info(1);
sun_ob = orbit_info(2:end);